### PR TITLE
add support for empty array item

### PR DESCRIPTION
**Pull request by profiles**

### CODE REVIEW

File: lib/evaluate.ts [hunk 1]
```diff
@@ -116,29 +116,34 @@ type Visitor<T> = (element: T, c: Continuation, cerr: PartialErrorContinuation)
...
-              accumulated = accumulated.slice(0, index);
-            }
-            accumulated.push(value);
-            visited.add(index);
-            schedule(() => loop(index + 1, accumulated));
-          },
-          cerr
-        );
+          //If true, it means an empty item like `[,]`
+          if(items[index] === null){
```
Comment (bartosz-k) [inline lib/evaluate.ts:126]: Will it work with arrays including `null` for real, like `[null]` etc?

Comment (profiles) [reply lib/evaluate.ts:126]: yes, beause it is an ast null, not a value null

Comment (bartosz-k) [reply lib/evaluate.ts:126]: Right... but `visitArray` is not right place to do that, it should be general as it is. It shouldn't check for nulls.

Comment (profiles) [reply lib/evaluate.ts:126]: as far as i know, ONLY empty array item will cause a null ast node, in fact the null ast node is illogicality, but all ast parser will generate a null ast node when parse an array with empty items

Comment (bartosz-k) [reply lib/evaluate.ts:126]: `visitArray` is an utility function, it's not created only to support AST nodes (including arrays nodes). Logic for nulls in AST nodes should be handled in ArrayExpression. I think we should map nulls into some special ast node like `NullValue`.
I can take care of it in one of future improvements. I tend to create one huge PR with many improvements, I can include your fix there written in a way I've described.

**Comment by bartosz-k** [comment]
Improvements within this PR will be included in the future in other changeset. Thanks @profiles .

**Comment by bartosz-k** [comment]
@profiles FYI, that's more proper way for MetaES: https://github.com/metaes/metaes/commit/5baad0605f399e15f483ce246437d5c956ecf25d.

**Pull request closed without merging**